tests/rectilinear_domain.cpp

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+
+#include <type_traits>
+
+#include <ddc/Coordinate>
+#include <ddc/DiscreteCoordinate>
+#include <ddc/DiscreteDomain>
+#include <ddc/NonUniformDiscretization>
+#include <ddc/RectilinearDomain>
+#include <ddc/UniformDiscretization>
+
+#include <gtest/gtest.h>
+
+namespace {
+
+struct X;
+struct Y;
+
+using DDimX = UniformDiscretization<X>;
+using DDimY = UniformDiscretization<Y>;
+using NUDDimX = NonUniformDiscretization<X>;
+using NUDDimY = NonUniformDiscretization<Y>;
+
+using DDomNull = DiscreteDomain<nullptr_t>;
+using DDomX = DiscreteDomain<DDimX>;
+using NUDDomX = DiscreteDomain<NUDDimX>;
+using DDomXY = DiscreteDomain<DDimX, DDimY>;
+using NUDDomXY = DiscreteDomain<NUDDimX, NUDDimY>;
+
+} // namespace
+
+TEST(RectilinearDomainTest, NonDiscreteDomainSpecialization_Value)
```

Comment:
```suggestion
TEST(RectilinearDomainTest, NonDiscreteDomainSpecializationValue)
```
Not critical but I think we are not allowed to use underscores in the naming of google tests.